src/Tools/ConfigurationSchemaGenerator/DiagnosticDescriptors.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.DotnetRuntime.Extensions;
+
+namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration;
+
+internal static class DiagnosticDescriptors
+{
+    private static readonly string s_projectName = typeof(DiagnosticDescriptors).Assembly.GetName().Name!;
+
+    public static DiagnosticDescriptor TypeNotSupported { get; } = CreateTypeNotSupportedDescriptor("TypeNotSupported");
+    public static DiagnosticDescriptor MissingPublicInstanceConstructor { get; } = CreateTypeNotSupportedDescriptor("SR.MissingPublicInstanceConstructor");
+    public static DiagnosticDescriptor CollectionNotSupported { get; } = CreateTypeNotSupportedDescriptor("SR.CollectionNotSupported");
+    public static DiagnosticDescriptor DictionaryKeyNotSupported { get; } = CreateTypeNotSupportedDescriptor("SR.DictionaryKeyNotSupported");
+    public static DiagnosticDescriptor ElementTypeNotSupported { get; } = CreateTypeNotSupportedDescriptor("SR.ElementTypeNotSupported");
+    public static DiagnosticDescriptor MultipleParameterizedConstructors { get; } = CreateTypeNotSupportedDescriptor("SR.MultipleParameterizedConstructors");
+    public static DiagnosticDescriptor MultiDimArraysNotSupported { get; } = CreateTypeNotSupportedDescriptor("SR.MultiDimArraysNotSupported");
+    public static DiagnosticDescriptor NullableUnderlyingTypeNotSupported { get; } = CreateTypeNotSupportedDescriptor("SR.NullableUnderlyingTypeNotSupported");
+
+    public static DiagnosticDescriptor PropertyNotSupported { get; } = new DiagnosticDescriptor(
+        id: "SYSLIB1101",
+        title: "PropertyNotSupportedTitle",
```

Comment:
I'm assuming we'll want to document these somewhere in the repo as well?

Reply:
These are "mocks" of the dotnet/runtime Configuration.Binder source generator's Diagnostics. I couldn't use the real code because it uses localized strings, which means we would need to get the "SR" generation working in the dotnet/aspire repo.

I'll add a comment about this.